internal/scorecard/scorecard.go

```diff
@@ -97,20 +96,13 @@ func (s Config) RunTests() error {
 		}
 	}
 
-	// Update the state for the tests
-	for _, suite := range pluginOutputs {
-		for idx, res := range suite.Results {
-			suite.Results[idx] = schelpers.UpdateSuiteStates(res)
-		}
-	}
-
 	if err := s.printPluginOutputs(pluginOutputs); err != nil {
 		return err
 	}
 
 	for _, scorecardOutput := range pluginOutputs {
 		for _, result := range scorecardOutput.Results {
-			if result.Fail > 0 || result.PartialPass > 0 || result.Error > 0 {
+			if result.State == scapiv1alpha2.FailState || len(result.Errors) > 0 {
```

Comment:
Does this cover all the cases (any failed or errored tests will mean we exit 0)?
```suggestion
			if result.State != scapiv1alpha2.PassState {
```